Firestore/core/src/firebase/firestore/nanopb/reader.cc

```diff
@@ -47,6 +53,31 @@ void Reader::ReadNanopbMessage(const pb_field_t fields[], void* dest_struct) {
   }
 }
 
+// GrpcByteBufferReader
+
+GrpcByteBufferReader::GrpcByteBufferReader(const grpc::ByteBuffer& buffer) {
+  std::vector<grpc::Slice> slices;
```

Comment:
Unfortunately, gRPC doesn't allow to avoid copying here (at least, I'm not aware of a way to achieve that). Since a copy has to be made, it seemed to make sense to have this reader own it rather than burdening the caller with the conversion.